src/GridFS/Bucket.php

```diff
@@ -0,0 +1,198 @@
+    /**
+     * Constructs a GridFS bucket.
+     *
+     * Supported options:
+     *
+     *  * bucketName (string): The bucket name, which will be used as a prefix
+     *    for the files and chunks collections. Defaults to "fs".
+     *
+     *  * chunkSizeBytes (integer): The chunk size in bytes. Defaults to
+     *    261120 (i.e. 255 KiB).
+     *
+     *  * readPreference (MongoDB\Driver\ReadPreference): Read preference.
+     *
+     *  * writeConcern (MongoDB\Driver\WriteConcern): Write concern.
+     *
+     * @param Manager  $manager      Manager instance from the driver
+     * @param string   $databaseName Database name
+     * @param array    $options      Bucket options
+     * @throws InvalidArgumentException
+     */
+    public function __construct(Manager $manager, $databaseName, array $options = [])
+    {
+        $collectionOptions = [];
+        $options += [
+            'bucketName' => 'fs',
+            'chunkSizeBytes' => 261120,
+        ];
+        if (isset($options['bucketName']) && ! is_string($options['bucketName'])) {
+            throw new InvalidArgumentTypeException('"bucketName" option', $options['bucketName'], 'string');
+        }
+        if (isset($options['chunkSizeBytes']) && ! is_integer($options['chunkSizeBytes'])) {
+            throw new InvalidArgumentTypeException('"chunkSizeBytes" option', $options['chunkSizeBytes'], 'integer');
+        }
+        if (isset($options['readPreference'])) {
+            if (! $options['readPreference'] instanceof ReadPreference) {
+                throw new InvalidArgumentTypeException('"readPreference" option', $options['readPreference'], 'MongoDB\Driver\ReadPreference');
+            } else {
+                $collectionOptions['readPreference'] = $options['readPreference'];
+            }
+        }
+        if (isset($options['writeConcern'])) {
+            if (! $options['writeConcern'] instanceof WriteConcern) {
+                throw new InvalidArgumentTypeException('"writeConcern" option', $options['writeConcern'], 'MongoDB\Driver\WriteConcern');
+            } else {
+                $collectionOptions['writeConcern'] = $options['writeConcern'];
+            }
+        }
+        $this->databaseName = (string) $databaseName;
+        $this->options = $options;
+
+        $this->filesCollection = new Collection(
+            $manager,
+            sprintf('%s.%s.files', $this->databaseName, $options['bucketName']),
+            $collectionOptions
+        );
+        $this->chunksCollection = new Collection(
+            $manager,
+            sprintf('%s.%s.chunks', $this->databaseName, $options['bucketName']),
+            $collectionOptions
+        );
+    }
+    /**
+     * Opens a Stream for reading the contents of a file specified by ID.
+     *
+     * @param ObjectId $id
+     * @return Stream
+     */
+    public function openDownloadStream(ObjectId $id)
+    {
+        fopen('gridfs://$this->databaseName/$id', 'r');
```

Comment:
I assume you just haven't gotten around to it yet, but this likely needs a context passed in for the Manager and/or Bucket instance.